Makefile

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8; mode: makefile-gmake -*-
+
+.DEFAULT = help
+TEST ?= .
+
+# python version to use
+PY       ?=3
+# list of python packages (folders) or modules (files) of this build
+PYOBJECTS = tokenlib
+# folder where the python distribution takes place
+PYDIST   ?= dist
+# folder where the python intermediate build files take place
+PYBUILD  ?= build
+
+SYSTEMPYTHON = `which python$(PY) python | head -n 1`
+VIRTUALENV   = virtualenv --python=$(SYSTEMPYTHON)
+VENV_OPTS    = "--no-site-packages"
+TEST_FOLDER  = ./tokenlib/tests
+
+ENV     = ./local/py$(PY)
+ENV_BIN = $(ENV)/bin
+
+
+PHONY += help
+help::
+	@echo  'usage:'
+	@echo
+	@echo  '  build     - build virtualenv ($(ENV)) and install *developer mode*'
+	@echo  '  lint      - run pylint within "build" (developer mode)'
+	@echo  '  test      - run tests for all supported environments (tox)'
+	@echo  '  dist      - build packages in "$(PYDIST)/"'
+	@echo  '  pypi      - upload "$(PYDIST)/*" files to PyPi'
```

Comment:
nit: IIUC this is now `make publish` rather than `make pypi`.

Reply:
Argh, I'am fool, I will fix it. Thanks for pointing out this.